espnet/nets/st_interface.py

```diff
@@ -0,0 +1,31 @@
+"""ST Interface module."""
+
+from espnet.nets.asr_interface import ASRInterface
+
+
+class STInterface(ASRInterface):
```

Comment:
If you just wanna rename `recognize` to `translate`. STInterface should transfer them
```python
class STInterface(ASRInterface):
    def recognize(self, x, trans_args, char_list=None, rnnlm=None):
        return self.translate(self, x, trans_args, char_list, rnnlm)

    def recognize_batch(self, x, trans_args, char_list=None, rnnlm=None):
        return self.translate_batch(self, x, trans_args, char_list, rnnlm)

    ...
```

If not so, STInterface should not be derived from ASRInterface. This is because they are not `is-a and we cannot cast STInterface to STInterface.

Reply:
I don't use recognize* now, so this is not what I wanna do.
Also, I want to keep both recognize* and translate* for both ST and ASR branches when we use multi-task learning (ST+ASR).

Reply:
OK. I understand. Can you describe this hierarchies in the Notes section of docstrings?